hledger/Hledger/Cli/Commands/Run.hs

```diff
@@ -38,41 +49,70 @@ runmode = hledgerCommandMode
   )
   cligeneralflagsgroups1
   hiddenflags
-  ([], Just $ argsFlag "[COMMANDS_FILE1 COMMANDS_FILE2 ...]")
+  ([], Just $ argsFlag "[COMMANDS_FILE1 COMMANDS_FILE2 ...] OR [command1 args... -- command2 args... -- command3 args...]")
 
--- | The fake run command introduced to break circular dependency
-run' :: CliOpts -> Journal -> IO ()
-run' _opts _j = return ()
+replmode = hledgerCommandMode
+  $(embedFileRelative "Hledger/Cli/Commands/Repl.txt")
+  (
+  []
+  )
+  cligeneralflagsgroups1
+  hiddenflags
+  ([], Nothing)
+
+-- | The fake run/repl command introduced to break circular dependency.
+-- This module needs access to `findBuiltinCommand`, which is defined in Hledger.Cli.Commands
+-- However, Hledger.Cli.Commands imports this module, which creates circular dependency.
+-- We expose this do-nothing function so that it could be included in the list of all commands inside
+-- Hledger.Cli.Commands and ensure that "run" is recognized as a valid command by the Hledger.Cli top-level
+-- command line parser. That parser, however, would not call run'. It has a special case for "run", and
+-- will call "run" (see below), passing it `findBuiltinCommand`, thus breaking circular dependency.
+runOrReplStub :: CliOpts -> Journal -> IO ()
+runOrReplStub _opts _j = return ()
+
+-- | Default input files + InputOpts that would be used by commands if
+--   there is no explicit alternative given
+newtype DefaultRunJournal = DefaultRunJournal (NE.NonEmpty String)
```

Comment:
Not a release blocker, but I think we'll be able to make this clearer. 
It's not a journal, it's a list of file paths.
What does "+ InputOpts" mean here ? I thought maybe it meant "also includes options like --infer-equity" but it seems not.

Reply:
PS I found it useful to derive Show here for debugging, might be worth adding.

Reply:
This is an obsolete comment, sorry :( I'll clean this up